### PR TITLE
feat: revamp add collection

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**

- Adds new create collection modal
- Adds component to add metadata to an entity

Files to review
- `templates/add-collection-modal`
- `organisms/add-metadata`
- `domains/collections/index`

image

### CODE REVIEW

File: src/domain/collections/index.js [hunk 1]
```diff
@@ -18,9 +18,51 @@ import {
...
 const CollectionsIndex = () => {
+  const [showNew, setShowNew] = useState(false)
+  const createCollection = useAdminCreateCollection()
+  const toaster = useToaster()
+  const { collections, isLoading, refresh, total_count } = useMedusa(
+    "collections",
+    {
+      search: `?${qs.stringify(filtersOnLoad)}`,
+    }
+  )
```
Comment (olivermrbl) [inline src/domain/collections/index.js:35]:
```suggestion
  const { collections, isLoading, count } = useAdminCollections()
```

File: src/domain/collections/index.js [hunk 2]
```diff
@@ -18,9 +18,51 @@ import {
...
+        }
+      }, {})
+
+      payload.metadata = metadata
+    }
+
+    createCollection.mutate(payload, {
+      onSuccess: () => {
+        toaster("Collection created", "success")
+        refresh()
```
Comment (olivermrbl) [inline src/domain/collections/index.js:57]: Should not be necessary, if we use the hook

**Review by olivermrbl — COMMENTED** [review]
Great work @kasperkristensen! LGTM, but just had a comment. Let me know 👍

File: src/domain/collections/index.js (hunk 2 above)
Comment (kasperkristensen) [reply src/domain/collections/index.js:57]: Agree but the collection view is being moved to the product overview in another PR and this domain will be deleted so this is just temporary 😄

**Pull request merged**